messaging/relayer/internal/messages/message_v1.go

```diff
@@ -0,0 +1,96 @@
+package messages
+
+/*
+ * Copyright 2019 ConsenSys Software Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// MessageV1 implements message interface.
+type MessageV1 struct {
```

Comment:
what do you think about having a namespacing approach to versioning the types, as opposed to embedding it in the type name? i.e. structuring things so that we would end up referring to the types as `v1.Message`, `v1.Proof` etc.

Do we want to follow semver for our data model? If so, does capturing only the major version here suffice, or should we capture at least the minor version. Does this pollute the type names if we use the current approach?

Reply:
Currently, our `XxxV1` refers to the message version "0.0.1" as opposed to what that naming might allude i.e. "1.0.0"

Reply:
MessageV1 follows Version1 from the versions.go and the Version1 constant can have its own naming scheme. What do you think? Structuring things looks fine to me

Reply:
Yea, I think if there are discrepancies between the two it will likely lead to confusion down the line. We can keep things as-is for now and see how we go I guess. We can refactor later if required.